include/revng/PipeboxCommon/ObjectID.h

```diff
@@ -170,6 +182,43 @@ class ObjectID {
     }
   }
 
+  std::vector<uint8_t> serializeBytes() const {
+    const Kind &TheKind = kind();
+    std::vector<uint8_t> Result;
+
+    if (TheKind == Kinds::Binary) {
```

Comment:
Here you are making a deep comparison. I'd use a comparison between pointers. They are singletons.